homeassistant/components/elevenlabs/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for ElevenLabs text-to-speech integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from elevenlabs.client import AsyncElevenLabs
+from elevenlabs.core import ApiError
+from elevenlabs.types import Model, Voice
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_KEY
+
+from .const import CONF_MODEL, CONF_VOICE, DOMAIN
+
+STEP_USER_DATA_SCHEMA_NO_AUTH = vol.Schema({vol.Required(CONF_API_KEY): str})
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ElevenLabsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for ElevenLabs text-to-speech."""
+
+    VERSION = 1
+
+    user_info: dict[str, Any] | None = None
+    voices: list[str] | None = None
+    models: list[str] | None = None
+
+    async def _get_voices_models(self, user_input: dict[str, Any]) -> None:
+        client = AsyncElevenLabs(api_key=user_input[CONF_API_KEY])
+        voices_async = client.voices.get_all()
+        models_async = client.models.get_all()
+        voices: list[Voice] = (await voices_async).voices
+        models: list[Model] = await models_async
+        voice_names = [voice.name if voice.name else "Unknown" for voice in voices]
```

Comment:
This might end up with 2 voices being called "Unknown". Can you use their ID instead?